### ISSUE TITLE
RegExp constructor doesn't support RegExp arguments

### ISSUE THREAD

**Issue by zloirock** [ota-meshi/eslint-plugin-regexp#258]
```js
const orig = /^https?:\/\//i;
const clone = new RegExp(orig); // => '^' will always reject because it is preceded by a character  regexp/no-useless-assertions
```

**Comment by RunDevelopment** [comment]
Our plugin currently [assumes that the first argument is a string](https://github.com/ota-meshi/eslint-plugin-regexp/blob/edb65997805f05ccfc2deffa639dc3c14b47ed44/lib/utils/index.ts#L352) and will simply convert the regex into a string. The RegExp constructor in your example basically gets (incorrectly) interpreted as `new RegExp("/^https?:\/\//i")`. 

The problem isn't with `no-useless-assertions` but that we don't support the RegExp constructor correctly.

@ota-meshi I'm not quite sure how to fix this. Do you want to take this one?
